### PR TITLE
Remove OpenSSL dependencies

### PR DESCRIPTION
**Pull request by LKozlowski**
This PR completely removes OpenSSL dependency from teleport binary.

It combines two things:
1) use of rdp-rs library without OpenSSL dependency (it is using rustls)
2) https://github.com/gravitational/teleport/pull/8914 

Also, I've added docs sections for setting up certificate templates for RDP connection

### CODE REVIEW

**Review by xacrimon — APPROVED** [review]

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:328]: Thanks for suggestion!

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:317]: Yes, you're right. I've added a sentence to the admonition at the beginning of that chapter that states that it is only necessary for Windows Server 2012 R2. Do you think it is enough or should I add it more explicitly?

Comment (zmb3) [reply docs/pages/desktop-access/getting-started.mdx:317]: @ptgott can advise us on this

File: lib/srv/desktop/rdp/rdpclient/Cargo.toml
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/Cargo.toml:23]: Don't merge this PR until the rdp-rs changes land in master, then update the `rev` here.

Comment (LKozlowski) [reply lib/srv/desktop/rdp/rdpclient/Cargo.toml:23]: No worries! That was my plan to wait until we finalize the rdp-rs changes :)

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:323]:
```suggestion
  This step is only *necessary* for Windows Server 2012 R2 as it does not support
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:326]: nit:

```suggestion
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:330]: nit: Let's pick a line length and use it consistently. Whatever the rest of the file does.

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:338]: Maybe:

```suggestion
1. Open the Microsoft Management Console (MMC):

    ```
    Start > Control Panel > Administrative Tools > Certificate Authority
    ```
```

so we avoid the line breaks, the way it is it'll probably mess with the list items.

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:340]: Possibly?

```suggestion
3. Find the `Computer` template on the list, right-click on it, then select `Duplicate Template`.
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:342]: Indent items inside the list to avoid breaking it. I would also space the contents, keeping the lines together breaks some markdown parsers.

```suggestion
4. In the `Compatibility` tab change `Certification Authority` to `Windows Server 2012 R2` and click `OK`.

    In the same tab change `Certificate recipient` to `Windows Server 2012 R2` and click `OK`.

```

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:332]: Let's give a brief description of what we are trying to achieve here, before the steps. Ie: "we need to create a certificate template with parameters X and Y".

Knowing the intent can be helpful if the UI changes, or if someone is doing the setup in a slightly different way. It also highlights what is required from steps we just have to do to get there.

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:344]: Does it have to be P384/SHA384 or did you pick those because they are the best available? Is there a lower bound?

No need to change the text, I'm just curious. Although, if there is a lower bound, let's mention it in the description that precedes this (ie, "certificates must be P384 or better").

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:347]: > (...) Change the object identifier to `1.3.6.1.4.1.311.54.1.2.`

What does `1.3.6.1.4.1.311.54.1.2.` mean? Could you add its "human" name in parenthesis, so people understand better what they are doing?

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:355]: Space and indent item contents.

```suggestion
13. Go back to the Certificate Authority window and right-click on `Certificate Templates`. Then:

    ```
    New > Certificate Template to Issue
    ```

    Select `RemoteDesktopAccess` and click `OK`.
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:357]: What does GPO mean? Could you define the term the first time it is mentioned? (Step 2/6 I think.)

Ie, "blabla GPO (Gnome Party Onslaught)" or whatever it is.  :)

File: docs/pages/desktop-access/getting-started.mdx
Comment (codingllama) [inline docs/pages/desktop-access/getting-started.mdx:370]: @zmb3 do you have a Windows in hand? Could you give the guide a try?

(Or anyone else that happens to be in the review, really.)

File: lib/service/service.go
Comment (codingllama) [inline lib/service/service.go:861]: Please run goimports or gofmt here. Indent looks wonky.

Comment (codingllama) [inline lib/service/service.go:861]:
```suggestion
                // authentication. Returns an error if the user attempts to start Desktop
```

**Review by codingllama — COMMENTED** [review]
Seems reasonable to me, assuming the functionality was manually tested. Thanks for doing this work!

> This PR completely removes OpenSSL dependency from Teleport's binaries.

Sadly, this isn't entirely true. `libfido2` pulls `libcrypto` into our tsh builds, but I suppose it is true now for RDP. Please edit the description to mention that this is related to RPD, it may give the wrong impression to people otherwise.

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:338]: Yeah this looks way better, thanks!

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:347]: This number is object identifier (OID) for Remote Desktop Access enhanced key usage extension, but this point will be removed completely. It turns out that we actually don't need to set it to make RDP working. The suggestion was good anyway. The human readable name should've been added if this point had to remain.

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:344]: Yeah, it have to be this one. I picked it as it is enabled be default by Windows Server 2012 R2 and is also supported by rustls library. The other one that would also work is P521/SHA384 but it would require user to enable it manually (few extra steps that doesn't provide much of difference in our case).

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:357]: GPO means Group Policy Object. We mention about it for the first time in step 2/6. It is also described at this step and then in the following steps we only use the abbreviation. Do you think the abbreviation should be explained here as well?

Comment (codingllama) [reply docs/pages/desktop-access/getting-started.mdx:357]: I missed the explanation in step 2. It would mention it there in the first occurrence instead of down the line, but I think the text is good here.

File: docs/pages/desktop-access/getting-started.mdx
Comment (zmb3) [reply docs/pages/desktop-access/getting-started.mdx:317]: @LKozlowski it would also be good to confirm that if you make these changes and try to run with an old (openssl) version of Teleport that it still works.

Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:317]: @zmb3 it works on both version with new certificates

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx]: thanks for suggestions!

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx]: Sure, I've updated it 4-space indents.

**Comment by LKozlowski** [comment]
@codingllama I've reverted that indentation and slightly changed one item in the list to remove all the docs lint warnings. Also, now it is exactly as other lists and texts in this file, so I'm assuming everything will render correctly.

**Review by zmb3 — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:317]: It looks like this is the final subsection of "Step 3/6: Configure a GPO to allow Teleport connections". I'm not sure that's the intention, though, since the second paragraph begins with "In this step we'll...". 

If this is the beginning of a new step, we'll need to promote this section heading to an H2 (using `##` instead of `###`), add "Step n/d:" to the title, and renumber the rest of the H2 section headings (e.g., this would be "Step 4/7").

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:324]: If we're advising some users to skip this step, I think we should add a link to the next step here.

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:321]: Is "a Windows server" one that we've referred to earlier in this guide? Is it the one where we've been configuring our service account and GPOs? We might want to specify which host commands in this section will be run on.

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:330]:
```suggestion
### Create a certificate template
```
Guessing this should be an H3 if the previous section was meant to be an H2.

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:332]:
```suggestion
In this section, we will create a certificate template that uses elliptic curve P-384 and uses SHA384 as the signature algorithm.
```
To distinguish the beginning of this paragraph from the beginning of the last paragraph. The reader may be wondering, "when does the step begin?"

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:345]:
```suggestion
6. Go to the `General` tab and change `Template display name` to `RemoteDesktopAccess`. Make sure `Template name` is also `RemoteDesktopAccess`.
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:350]:
```suggestion
11. Finally, create a template by clicking `OK`.
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:358]:
```suggestion
### Update GPO to use a new certificate template
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:321]: Yes this is the same Windows server. I'll add a clarification.

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:324]: Great suggestion! It'll make it a bit easier for users to follow the steps in our guide.

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx:317]: I was thinking about whether I should add it as a separate step or add it to the existing one as it's related to it. After I gave it some thought and re-read it again, I agree that it make more sense to make it a separate optional step.

File: docs/pages/desktop-access/getting-started.mdx
Comment (LKozlowski) [reply docs/pages/desktop-access/getting-started.mdx]: Is there a way to know what will be the link to the header? I see that the links are auto-generated based on the text, so I went to the current live docs and took it from there. I was wondering if there is a better way to do it?

Comment (codingllama) [reply docs/pages/desktop-access/getting-started.mdx]: That's what I do too. If you wanted to be explicit you could stick in some plain HTML  (`<h2 id="myanchor">Text</h2>`), but I think the preference is for markdown.

Comment (codingllama) [reply docs/pages/desktop-access/getting-started.mdx]: Btw, you can render locally using https://github.com/gravitational/next - either checkout or create your branch in the submodule within the sources.

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:143]:
```suggestion
## Step 3/7. Configure a GPO to allow Teleport connections
```

For consistency with other steps

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:325]:
```suggestion
## Step 4/7. Configure a certificate for RDP connections
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:355]:
```suggestion
8. Next, in the `Extensions` tab select `Application Polices` and click the `Edit` button.
```

File: docs/pages/desktop-access/getting-started.mdx
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:357]:
```suggestion
10. Go to the `Security` tab, select `Domain Controllers` and give the group `Read` and `Enroll` permissions.
```

**Review by ptgott — APPROVED** [review]
Approved with some very minor suggestions

**Review by jakule — APPROVED** [review]

**Pull request merged**
